admin_guide/configuring_authentication.adoc

```diff
@@ -0,0 +1,526 @@
+
+.Master Configuration Using `GitHubIdentityProvider`
+====
+
+----
+oauthConfig:
+  ...
+  identityProviders:
+  - name: github <1>
+    challenge: false <2>
+    login: true <3>
+    provider:
+      apiVersion: v1
+      kind: GitHubIdentityProvider
+      clientID: ... <4>
+      clientSecret: ... <5>
+----
+<1> This provider name is prefixed to the GitHub numeric user ID to form an identity name. It is also used to build the callback URL.
+<2> `GitHubIdentityProvider` cannot be used to send `WWW-Authenticate` challenges.
+<3> When `true`, unauthenticated token requests from web clients (like the Management Console) are redirected to GitHub to log in.
+<4> The client ID of a link:https://github.com/settings/applications/new[registered GitHub OAuth application]. The application must be configured with a callback URL of `_<master>_/oauth2callback/_<identityProviderName>_`.
+<5> The client secret issued by GitHub.
+====
+
+=== Google [[Google]]
+
+Set `GoogleIdentityProvider` in the `identityProviders` stanza to use Google as
+an identity provider, using
+https://developers.google.com/identity/protocols/OpenIDConnect[Google's OpenID
+Connect integration].
+
+NOTE: Using Google as an identity provider requires users to get a token using
+`_<master>_/oauth/token/request` to use with command-line tools.
+
+.Master Configuration Using `GoogleIdentityProvider`
+====
+
+----
+oauthConfig:
+  ...
+  identityProviders:
+  - name: google <1>
+    challenge: false <2>
+    login: true <3>
+    provider:
+      apiVersion: v1
+      kind: GoogleIdentityProvider
+      clientID: ... <4>
+      clientSecret: ... <5>
+----
+<1> This provider name is prefixed to the Google numeric user ID to form an identity name. It is also used to build the redirect URL.
+<2> `GoogleIdentityProvider` cannot be used to send `WWW-Authenticate` challenges.
+<3> When `true`, unauthenticated token requests from web clients (like the Management Console) are redirected to Google to log in.
+<4> The client ID of a link:https://console.developers.google.com/[registered Google project]. The project must be configured with a redirect URI of `_<master>_/oauth2callback/_<identityProviderName>_`.
+<5> The client secret issued by Google.
+====
+
+=== OpenID Connect [[OpenID]]
+
+Set `OpenIDIdentityProvider` in the `identityProviders` stanza to integrate with
+an OpenID Connect identity provider using an
+link:http://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth[Authorization Code Flow].
+
+NOTE: *ID Token* and *UserInfo* decryption is not supported.
+
+By default, the `openid` scope is requested. Extra scopes, if needed, can be
+specified in the `extraScopes` field.
+
+Claims are read from the JWT `id_token` returned from the OpenID identity
+provider and, if specified, from the JSON returned by the `UserInfo` URL.
+
+At least one claim must be configured as the claim to use as the user's
+identity. The
```

Comment:
@adellape Suggestion: Sentence could probably be "At least one claim must be configured to use as the user's identity." Otherwise there's a little repetition.